src/explorer/connectionTreeItem.ts

```diff
@@ -67,21 +68,24 @@ export default class ConnectionTreeItem
     childrenCache: {
       [key: string]: DatabaseTreeItem | StreamProcessorTreeItem;
     }; // Existing cache.
+    isMutable: boolean;
   }) {
     super(
       connectionController.getSavedConnectionName(connectionId),
       collapsibleState
     );
 
-    if (
+    const isConnected =
       connectionController.getActiveConnectionId() === connectionId &&
       !connectionController.isDisconnecting() &&
-      !connectionController.isConnecting()
-    ) {
-      this.contextValue = ConnectionItemContextValues.connected;
-    }
+      !connectionController.isConnecting();
+
+    this.contextValue = `${isConnected ? 'connected' : 'disconnected'}${
+      isMutable ? '' : 'Immutable'
```

Comment:
naming here I'm not so sure about. Immutable could also be "Preset", I just thought of making this more generic. The idea is that it's not a connection that can be removed/edited/etc.